### PR TITLE
refactor(edgeless): element tree manager

### PR DESCRIPTION
**Pull request by L-Sun**
### TL;DR
This PR introduces `TreeManager` to maintain the container-children relationship among `GfxModel`s. This change can make sure an element only has zero or one container. 

### What changes:
- A new `TreeManager` is introduced to maintain the container-children relationship.  Its implementation references the relevant code previously used in group in `surface-model.ts`. 
- Removed code related to the container-children relationship watcher of group.
- The `ContainerElement` interface and related code have been moved from `element-model.ts` to a new file, `container-element.ts`.
- Simplified the implementation of the `ContainerElement` interface in `GroupLikeModel` and `FrameModel`.
- Some type annotation changes: from `BlockSuite.EdgelessModel` to `GfxModel`.

### About test:
The existing tests already cover the changes mentioned above, including the relationship changes after operations on the child elements of frames and groups.

### CODE REVIEW

**Comment by L-Sun** [comment]
* **#8239** <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/8239?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @L-Sun and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/framework/block-std/src/gfx/surface/element-model.ts [hunk 1]
```diff
@@ -134,6 +144,10 @@ export abstract class GfxPrimitiveElementModel<
     return true;
   }
 
+  get container() {
```
Comment (doouding) [inline packages/framework/block-std/src/gfx/surface/element-model.ts:147]: What's the difference between group and container?

Comment (L-Sun) [reply packages/framework/block-std/src/gfx/surface/element-model.ts:147]: The contianer() can return block contianer like frame

**Review by Saul-Mirone — APPROVED** [review]
LGTM

**Comment by L-Sun** [comment]
### Merge activity

* **Oct 14, 5:18 AM EDT**: The merge label 'merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=toeverything&repo=blocksuite) once it meets the requirements.
